### PR TITLE
add http service

### PR DESCRIPTION
**Pull request by suixin567**
@Suprcode 
aim:
 Let the web server communicate with the game server
route:
1、/regist   regist a new account by way of website form.
2、/login  login website with game account
3、/addNameList  add a nameList by way of website
and there is a trusted ipaddress can be config , for security.

### CODE REVIEW

File: Server/Settings.cs [hunk 1]
```diff
@@ -59,6 +59,9 @@ internal static class Settings
                              RelogDelay = 50,
                              MaxIP = 5;
 
+        //HTTP
+        public static string HTTPIPAddress = "http://127.0.0.1:5679/";
+        public static string HTTPTrustedIPAddress = "127.0.0.1:5679";
```
Comment (Suprcode) [inline Server/Settings.cs]: what is this HTTPTrustedIPAddress used for? It has no implementation from what i can see.

File: Server/MirEnvir/Envir.cs [hunk 2]
```diff
@@ -526,7 +527,8 @@ private void WorkLoop()
                 }
 
                 StartNetwork();
-
+                http = new HttpServer();
```
Comment (Suprcode) [inline Server/MirEnvir/Envir.cs]: please put this under a config setting to toggle starting the httpserver

File: Server/Utils/HttpService.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            if (listener != null && listener.IsListening)
+            {
+                listener.Stop();
+            }           
+        }
+
+        public abstract void OnGetRequest(HttpListenerRequest request, HttpListenerResponse response);
+        public abstract void OnPostRequest(HttpListenerRequest request, HttpListenerResponse response);
+
+        public void WriteRresponse(HttpListenerResponse response, string responseString)
```
Comment (Suprcode) [inline Server/Utils/HttpService.cs]: how come its WriteRresponse, and not WriteResponse?

**Comment by Suprcode** [comment]
thanks. Just a question on the WriteRresponse naming. And still need a config setting to not start the http service.

Many people won't want this enabled so should be off by default.

**Comment by suixin567** [comment]
@Suprcode  
Toggle have been added, and default is off. 
the next my plan is :
1、support chanese character name
2、make game class can be off , because the assassins and archers are unfamiliar for most of us.
And I have a wondering about : how to make players do not need install a DirectX SDK ? if do not install ,there will a black screen... thanks very much.

**Comment by Suprcode** [comment]
You can already turn off archers and assassin in the code. This is in the config menu.

No need for built in DirectX. The 4 dll files needed are already supplied with the client. No need for sdk

**Comment by suixin567** [comment]
@Suprcode  sorry I do not understand , is there a toggle in config menu to off class ?  if so ,I don't have to do it again.
and the 4 dll files do not work ,  I tried it on two computers, both are black with that 4 dll files ~~~

**Comment by Suprcode** [comment]
If you're still having a black screen after the 4 dll's - then yes you do need to install it.

I expect its different depending on your hardware. I don't have to install the SDK on mine. The 4 dll's are the only extra ones i need.

In config menu -> Permissions

You can disable assassin and archer class already

File: Server/MirEnvir/Envir.cs [hunk 4]
```diff
@@ -526,7 +527,11 @@ private void WorkLoop()
                 }
 
                 StartNetwork();
-
+                if (Settings.StartHTTPService)
+                {
+                    http = new HttpServer();
```
Comment (Suprcode) [inline Server/MirEnvir/Envir.cs]: this instance is no longer required as you're creating it in the constructor

**Comment by suixin567** [comment]
@Suprcode sorry!  instance problem haved fixed ! I was careless

**Comment by Suprcode** [comment]
> @Suprcode sorry! instance problem haved fixed ! I was careless

I'm not able to see this change yet? Still creates 2 instances.

**Comment by suixin567** [comment]
@Suprcode  the one is removed from constructor.  the instance is only create when setting is on.

**Comment by suixin567** [comment]
forget to push.......

**Pull request merged**